packages/macro/src/utils.ts

```diff
@@ -0,0 +1,52 @@
+import * as t from "@babel/types";
+import { Options as RuntimeOptions } from "use-ssr-computation.runtime/src/utils"
+
+type MacroOptions = {
+  serverSideOnly?: boolean;
+  webpackChunkName?: string;
+};
+
+export type Options = MacroOptions & RuntimeOptions;
+
+// `ExtractObjectTypes` maps each property of the input object `T` to a string representing its type
+type ExtractObjectTypes<T> = {
```

Comment:
It ensures that if any new option is added to the `MacroOptions`, it must be added to all objects and functions depending on it. @Romex91 If you see it overcomplicating the code, please tell me.

Reply:
No, I don't see it as over-complication. This is absolutely necessary. Compile time errors are great!